animareflection/ui
# Commit linting

## Description

##### Task link: https://trello.com/c/O7zkMZIX/41-commit-linting-hook

- Added commit linting via [commitlint](https://commitlint.js.org/#/) with Husky-enabled hook (`commit-msg`)
- Updated lifecycle hook scripts to work for Yarn Berry (based on [Husky Yarn docs](https://typicode.github.io/husky/getting-started.html#yarn-2))
- Explicitly set introduction MDX as first in story ordering (see [this comment](https://github.com/animareflection/ui/pull/56/files#r1304610536))

## Test Steps

- Locally, test `commitlint` and make sure it leverages `commitlint.config.ts`
  - This can be tested by piping input to the `commitlint` CLI. For example, `echo 'test: example' | yarn commitlint`
- Verify the Husky `commit-msg` hook automatically executes `commitlint`
  - This can be tested by adding a random file (don't commit it, just `git add`), then testing various git messages with `git commit`
  - CLI & config unit tests will be added later once this config is extracted to a shared repo (checklist item added to https://trello.com/c/O7zkMZIX/41-commit-linting-hook scope)
- Verify the `commitlint.config.ts` config looks good, particularly the commit types
- Verify introduction story is loaded first on page load
